### ISSUE TITLE
[java] FieldNamingConventions false positive: anonymous inner class instance field inside interface default method

### ISSUE THREAD

**Issue by qualidafial** [pmd/pmd#1812]
**Affects PMD Version:** 6.10.0

**Rule:** FieldNamingConventions

**Description:** Instance fields of anonymous inner classes inside interface default methods are incorrectly flagged as needing CONSTANT_CASE naming.

**Code Sample demonstrating the issue:**

```java
package com.example;

import java.util.Iterator;
import java.util.function.Function;

@FunctionalInterface
public interface MappableIterable<T> extends Iterable<T> {
    default <U> Iterator<U> mapIterator(Function<T, U> mapper) {
        return new Iterator<U>() {
            private final Iterator<T> delegate = iterator();

            @Override
            public boolean hasNext() {
                return delegate.hasNext();
            }

            @Override
            public U next() {
                return mapper.apply(delegate.next());
            }
        };
    }
}
```

Output:

```
[INFO] --- maven-pmd-plugin:3.9.0:check (default) @ example ---
[INFO] PMD Failure: com.example.MappableIterable:23 Rule:FieldNamingConventions Priority:1 The public constant name 'delegate' doesn't match '[A-Z][A-Z_0-9]*'.
[INFO] PMD Failure: com.example.MappableIterable:23 Rule:FieldNamingConventions Priority:1 The public constant name 'delegate' doesn't match '[A-Z][A-Z_0-9]*'.
```

**Running PMD through:** *Maven* (maven-pmd-plugin version 3.9.0)

Fails both in TravisCI (running on openjdk 8) and on my workstation:

```bash
$ uname -a
Linux matt-hall-xps 4.15.0-48-generic #51-Ubuntu SMP Wed Apr 3 08:28:49 UTC 2019 x86_64 x86_64 x86_64 GNU/Linux
$ mvn -version
Apache Maven 3.6.0
Maven home: /usr/share/maven
Java version: 1.8.0_191, vendor: Oracle Corporation, runtime: /usr/lib/jvm/java-8-openjdk-amd64/jre
Default locale: en_US, platform encoding: UTF-8
OS name: "linux", version: "4.15.0-48-generic", arch: "amd64", family: "unix"
```

Note that this code does _not_ fail the build when I use openjdk 11.0.2.

**Comment by qualidafial** [comment]
Upgrading to 6.14.0 resolved issue
